scikit_mol/fingerprints.py

```diff
@@ -720,3 +726,238 @@ def parallel_helper(args):
 
     transformer = getattr(fingerprints, classname)(**parameters)
     return transformer._transform(X_mols)
+
+
+class FpsGeneratorTransformer(FpsTransformer):
+    _regenerate_on_properties = ()
+
+    def _fp2array(self, fp):
+        raise DeprecationWarning("Generators can directly return fingerprints")
+
+    def _mol2fp(self, mol):
+        raise DeprecationWarning("use _mol2array")
```

Comment:
> As you may have seen, the Master has moved, so likely there is a need to merge the updated master into the branch, and fix eventual conflicts that may arise. There's a little bit of error handling now, so that nonMols and other falsy objects return np.maskedarrays. We likely need to rework the stack _transform_mol, _fp_to_array and _mol_to_fp in the new generator class somewhat in the new abstract class. Likely the _mol2fp needs to be the abstract one, then _fp2array needs to be overriden in the new subclass, as we don't need to convert the array, but then this method can do the error handling for the conversion. Maybe the names need to be rethought at some point in the future.

Hej @EBjerrum ,
I'm a bit confused right now on the quote above, in the current generator approach we killed both `_fp2array` and `_mol2fp`.

We only have `_transform_mol` and `_generate_fp_generator` as abstracts. Are we missing something? Most likely the error handling, no?

Reply:
Yeah, hmm, maybe I described it wrong. Main thing is that we need to be able to pass back masked arrays if we get any object that is evaluated as False in a boolean context (None, Nan, InvalidMol etc). So there need to be some kind of function in the parent class that guards the abstract function, as the subclasses should not have to deal with that, but can expect the molecules to be valid.

How we organize the private functions used by the public transform method is of course up to us. Maybe the current parent class implementation is not really that smart when it comes to the new generator code and can be simplified.

Have you pushed your current work to the pull request? Then I can have a look at it.

Reply:
yes, in the current status everything should be included. :)